driver/node.go

```diff
@@ -211,6 +211,12 @@ func (s *nodeService) nodePublishFilesystemVolume(req *csi.NodePublishVolumeRequ
 		return status.Errorf(codes.Internal, "target device is already formatted with different filesystem: volume=%s, current=%s, new:%s", req.GetVolumeId(), fsType, mountOption.FsType)
 	}
 
+	// avoid duplicate XFS UUIDs
+	if mountOption.FsType == "xfs" {
```

Comment:
I think duplicate UUIDs also occur in EXT4; why is it that EXT4 does not need to be addressed? For examle, when duplicate UUIDs are exists, users can't specify the UUID in fstab.

Reply:
Thanks for bringing that up @cupnes ! Actually, no such issue is seen with `ext4` fsType and just to be double sure I have verified the same:
StorageClass:
```
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  creationTimestamp: "2022-04-23T08:24:12Z"
  name: topolvm-provisioner-ext4-thin
  resourceVersion: "55614"
  uid: 53f68d4b-4d20-43e5-b3c0-80284d343485
parameters:
  csi.storage.k8s.io/fstype: ext4
  topolvm.cybozu.com/device-class: ssd-thin
provisioner: topolvm.cybozu.com
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer
```
Then created its PVC-POD and then its Clone in the backend. 
```
NAME                     STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                    AGE

topolvm-pvc-ext4-clone   Bound    pvc-2ea7f693-8508-41a7-b758-ff9eea09a27e   2Gi        RWO            topolvm-provisioner-ext4-thin   3s
topolvm-pvc-ext4-thin    Bound    pvc-ac5ee4f0-17a6-4528-849f-fcc554937301   2Gi        RWO            topolvm-provisioner-ext4-thin   2m45s

```

Pods:
```
NAME                READY   STATUS    RESTARTS   AGE
my-pod-ext4-clone   1/1     Running   0          7s
my-pod-ext4-thin    1/1     Running   0          2m49s

```

`lvs`
```
  LV                                   VG    Attr       LSize Pool     Origin                               Data%  Meta%  Move Log Cpy%Sync Convert

  f19b9abb-2deb-4dd0-ac98-d40944cb2fcb myvg1 Vwi-aotz-- 2.00g thinpool                                      4.76                                   
  3c538661-b2b8-4def-b5a7-d6e01946aed6 myvg1 Vwi-aotz-- 2.00g thinpool f19b9abb-2deb-4dd0-ac98-d40944cb2fcb 4.76             
  thinpool                             myvg1 twi-aotz-- 2.00g                                               5.40   12.40                           
                      

```

Reply:
@Yuggupta27 Thank you. I understood.

Reply:
Have you confirmed that it works as expected by writing different data to both the original volume and the clone?
I think it would be safer to modify the UUID. At the very least, we should check with the ext4 team.

Reply:
Thanks for the suggestion @cupnes @nbalacha , I Updated the PR so that we generate a new UUID for both `xfs` and `ext4` ; just to be safe :)